tools/gfx/d3d12/d3d12-base.h

```diff
@@ -0,0 +1,73 @@
+// d3d12-base.h
+// Shared header file for D3D12 implementation
+#pragma once
+
+#include "../command-encoder-com-forward.h"
+#include "../d3d/d3d-swapchain.h"
+#include "../mutable-shader-object.h"
+#include "../renderer-shared.h"
+#include "../simple-render-pass-layout.h"
+#include "../transient-resource-heap-base.h"
+#include "core/slang-basic.h"
+#include "core/slang-blob.h"
+#include "core/slang-chunked-list.h"
+#include "descriptor-heap-d3d12.h"
```

Comment:
Maybe good to rename this to d3d12-descriptor-heap. We should check other files under d3d12 directory and make their naming consistent.

Reply:
Done for both descriptor heap and resource. I'm presuming swapchain and util are used for both D3D11 and D3D12 based on their current names so I left those alone.